src/engine/main.cpp

```diff
@@ -115,7 +115,7 @@ void quit()                  // normal exit
     exit(EXIT_SUCCESS);
 }
 
-volatile int errors = 0;
```

Comment:
I think the `volatile` keyword was used because the `fatal` function can be called from signal handlers, which the compiler can't predict, so we always need to get the latest value of `errors` from memory.

Reply:
But all `fatal` does is show up to 1 error window (more specifically `SDL_MESSAGEBOX_ERROR`) and write up to 2 fatal errors in log.txt, correct?

If so, what could be done is either:
* Show only 1 error in the log (the one printed by the window)
* Use `std::atomic<unsigned short>` or something to make the amount of fatal errors configurable for the log

Reply:
I can't see any current risks with removing `volatile` since the code inside `fatal` calls functions defined in other translation units while `errors` is not static, so the compiler can't make any assumptions about the access to `errors`, assuming link time optimization is not used, so `errors` is probably both written to and read from memory anyway. But I still think it's a good idea in general to make global or static variables accessed by signal handlers volatile just to be sure the compiler doesn't do anything unwanted.

Reply:
Can we just get rid of this variable entirely?

Reply:
The problem with getting rid of the variable is that then we can get recursive errors if a signal happens while the signal handler is executing.